Update Style Guide to include notes on CamelCase. (#1668)
UniMath/UniMath: UniMath/README.md

checking systems, and we follow a style of coding designed to render proofs
less fragile and to make the files have a more uniform and pleasing appearance.

* Identifiers and function names
  * Form identifiers by concatenating English words or existing identifiers in
    lower case, separating them by underscores.
  * Unless it impedes clarity or goes against common practice avoid using
    abbreviations.
  * In some parts of the library uppercase is used for bundled mathematical
    objects (e.g. `Pullback`, `Topos`).  It is sometimes justified to introduce
    new identifiers using this naming scheme.  The following guidelines should
    then be applied:
    * Identifiers with capital letters must not use underscores to separate
      words, they must use `CamelCase`.
    * Only use `CamelCase` when it is already used in the parts of the library
      you are working in or there is some compelling reason for it to be
      introduced.
    * Do not use `CamelCase` for intermediary structures.  Example: if
      `CamelCase`contains a data part and a property part then name these
      `camel_case_data` and `is_camel_case`, do not call them `CamelCaseData`
      and `IsCamelCase`.

>     * Capital letters must _never_ occur in function names.  Name your functions

Author: ```suggestion
    * Upper-case letters should not occur in function names.  Name your functions
```

>     * Capital letters must _never_ occur in function names.  Name your functions

Reviewer: I guess that sounds fine. Another option would be `Avoid capital letters in function names`, but I am unsure which one is better.

@benediktahrens Do you have suggestions
